homeassistant/components/aurora_abb_powerone/sensor.py

```diff
@@ -50,6 +51,7 @@ class AuroraABBSolarPVMonitorSensor(Entity):
     def __init__(self, client, name, typename):
         """Initialize the sensor."""
         self._name = f"{name} {typename}"
+        self._unique_id = f"{slugify(client.serline.port)}-{client.address}-{typename}"
```

Comment:
This is a com port and cannot be used as a unique ID .

Reply:
So just use the client address and the typename? Since the typename is different for every sensor, see https://github.com/home-assistant/home-assistant/pull/31649#issuecomment-583790323

Reply:
Or am I misunderstanding the following part of the documentation?

> If a device has a single serial but provides multiple entities, combine the serial with unique identifiers for the entities. For example, if a device measures both temperature and humidity, you can uniquely identify the entities using {serial}-{sensor_type}.

Thought I did the same with the use of the typename which represents the sensor type.

Reply:
If the issue with using the slugified com port is that it might overlap with something else, then I’m not sure how to guarantee a unique ID other than adding the component name as part of the unique ID (eg a prefix). 

There is no MAC address and we can’t guarantee that the device responds at HA startup (so we can’t read its serial number).

Reply:
> If the issue with using the slugified com port is that it might overlap with something else, then I’m not sure how to guarantee a unique ID other than adding the component name as part of the unique ID (eg a prefix).

I do not think that it can overlap with anything else since it is limited to this integration and this platform according to the [documentation](https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements). Pretty sure a combination of client.address and typename would be enough to fulfill all requirements of a unique_id since every other registered sensor would have a different typename so all of them would have their own indepent unique id.

Reply:
Ok, then that seems fine to me.

@balloob can confirm hopefully.

Reply:
If a the device is disconnected and a different device is connected using the same port as the disconnected device, there's no way of telling them apart, right? I think that's the problem Paulus is aiming at.

This is why we eg don't allow using ip addresses as unique_id.

Reply:
That makes a lot more sense, thanks for the explanation!
```python
self._unique_id = f"{client.address}-{typename}"
```
Without using the port it should be fine right?

Reply:
What is `client.address`?

Reply:
https://gitlab.com/energievalsabbia/aurorapy/-/blob/master/aurorapy/client.py#L534

`client.serial_number()` would be perfect but it is not available during the night, see https://github.com/home-assistant/home-assistant/pull/31649#issuecomment-586495192

Reply:
We can't guarantee that the serial line address is unique if there are more than one device, right?

Reply:
Yeah, you are right. So should I close this PR or is there any other option?

Reply:
Yeah, unfortunately, it looks like that's best for now. :+1: